core/integration/directory_test.go

```diff
@@ -318,6 +325,27 @@ func TestDirectoryWithFile(t *testing.T) {
 		File("sub-dir/target-file").Contents(ctx)
 	require.NoError(t, err)
 	require.Equal(t, "some-content", content)
+
+	dir := c.Directory().
```

Comment:
I think it'd be good to wrap this in a `t.Run("respects permissions", func(t *testing.T) {` so the test is less monolithic.

Reply:
👍🏻